IOSurfaceGetID is public API

rreno opened the pull request:
#### edc821880f856e6814483f6d5971b5f9cc9d4f1b
<pre>
IOSurfaceGetID is public API
<a href="https://bugs.webkit.org/show_bug.cgi?id=253973">https://bugs.webkit.org/show_bug.cgi?id=253973</a>
rdar://106762871

Reviewed by Tim Horton.

IOSurface::surfaceID was written before IOSurfaceGetID was available in
the public SDK on all platforms. This removes the preprocessor platform
checks and simply wraps IOSurfaceGetID. This also removes the
now-redundant IOSurfaceID typedef from the SPI header.

* Source/WebCore/platform/graphics/cocoa/IOSurface.mm:
(WebCore::IOSurface::surfaceID const):
* Source/WTF/wtf/spi/cocoa/IOSurfaceSPI.h:

Canonical link: <a href="https://commits.webkit.org/261732@main">https://commits.webkit.org/261732@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ef0e6aea8e29236f6009070502136f362c29eeb9

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/112633 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/21787 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/90/builds/1298 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/4409 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/121167 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/116700 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/23127 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/12952 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/86/builds/5561 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/118401 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/17181 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/100402 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/105708 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/99134 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/88/builds/948 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/46200 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/100927 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/14117 "Built successfully") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/89/builds/987 "1 flakes 6 failures") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/95307 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/12224 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/14798 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/10343 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/102416 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/20134 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/52982 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/31984 "Passed tests") | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/8182 "Failed to push commit to Webkit repository") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/16640 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/110457 "Built successfully") | 
| | | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/27284 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on Source/WebCore/platform/graphics/cocoa/IOSurface.mm:
```diff
@@ -469,13 +469,7 @@ static IntSize computeMaximumSurfaceSize()
-// FIXME: Should be able to do this on watchOS and tvOS.
-#if PLATFORM(IOS_FAMILY) && (!USE(APPLE_INTERNAL_SDK) || PLATFORM(WATCHOS) || PLATFORM(APPLETV))
-    return 0;
-#else
```
Is there cleanup to be done in the SPI header?

rreno replied:
There's an IOSurfaceID typedef for watch and tv in `WTF/wtf/spi/cocoa/IOSurfaceSPI.h`. Can probably remove that as well.

rreno committed:
IOSurfaceGetID is public API
https://bugs.webkit.org/show_bug.cgi?id=253973
rdar://106762871

Reviewed by Tim Horton.

IOSurface::surfaceID was written before IOSurfaceGetID was available in
the public SDK on all platforms. This removes the preprocessor platform
checks and simply wraps IOSurfaceGetID. This also removes the
now-redundant IOSurfaceID typedef from the SPI header.

* Source/WebCore/platform/graphics/cocoa/IOSurface.mm:
(WebCore::IOSurface::surfaceID const):
* Source/WTF/wtf/spi/cocoa/IOSurfaceSPI.h:

Canonical link: https://commits.webkit.org/261732@main
